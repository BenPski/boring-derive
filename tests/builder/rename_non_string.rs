use boring_derive::Builder;

#[derive(Debug, Default, Builder)]
struct Example {
    #[builder(rename = 1)]
    item: String,
}

fn main() {
    let ex = Example::default().value("val");
    println!("{:?}", ex);
}
