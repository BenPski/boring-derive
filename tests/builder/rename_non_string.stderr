error: rename must be a string not `1`
 --> tests/builder/rename_non_string.rs:5:15
  |
5 |     #[builder(rename = 1)]
  |               ^^^^^^^^^^

error[E0599]: no method named `value` found for struct `Example` in the current scope
  --> tests/builder/rename_non_string.rs:10:33
   |
4  | struct Example {
   | -------------- method `value` not found for this struct
...
10 |     let ex = Example::default().value("val");
   |                                 ^^^^^ method not found in `Example`
